pinot-query-runtime/src/main/java/org/apache/pinot/query/service/QueryDispatcher.java

```diff
@@ -208,8 +213,8 @@ public DispatchClient(String host, int port) {
       _blockingStub = PinotQueryWorkerGrpc.newBlockingStub(_managedChannel);
     }
 
-    public Worker.QueryResponse submit(Worker.QueryRequest request) {
-      return _blockingStub.submit(request);
+    public Worker.QueryResponse submit(Worker.QueryRequest request, long timeoutMs) {
+      return _blockingStub.withDeadlineAfter(timeoutMs, TimeUnit.MILLISECONDS).submit(request);
```

Comment:
Can we pass in withDeadline instead so they have the same deadline? 

Let's get this checked in and do async RPC in a separate PR.